src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -136,9 +136,9 @@ private class MethodCategory
             /// List of exception types which are allowed to be thrown inside this category of method.
             /// This list will be empty if no exceptions are allowed.
             /// </summary>
-            public ImmutableHashSet<ITypeSymbol> AllowedExceptions { get; }
+            public ImmutableHashSet<ITypeSymbol?> AllowedExceptions { get; }
```

Comment:
💭 Seems like this should not allow null elements.

```suggestion
            public ImmutableHashSet<ITypeSymbol> AllowedExceptions { get; }
```